src/main/webapp/js/selectScript.js

```diff
@@ -14,40 +14,40 @@
 export const CollegePlanner = (() => {
   let selected = [];  // Courses selected by the user
   let courses = [];   // List with all courses
+  let courseIdNameMap = {};  // JSON mapping of course ID {string} to course name {string}
```

Comment:
Try doing it this way:

```
/**
 * @type {{courseId: string}, {courseName: string}}
 */
const courseIdNameMap = {};
```

also consider `const courseNames = {};` - it might be clear enough. Up to you on the exact naming here.